sphinx/builders/linkcheck.py

```diff
@@ -125,16 +125,6 @@ def init(self) -> None:
         # set a timeout for non-responding servers
         socket.setdefaulttimeout(5.0)
 
-        # create queues and worker threads
-        self.rate_limits = {}  # type: Dict[str, RateLimit]
-        self.wqueue = queue.PriorityQueue()  # type: queue.PriorityQueue
-        self.rqueue = queue.Queue()  # type: queue.Queue
-        self.workers = []  # type: List[Thread]
```

Comment:
`rate_limits` was added at 6b90a63f08 (v3.4.0). So I think we don't need to pay for the compatibility.
On the other hand, `rqueue` and `wqueue` (at c3c879f2c6, since v1.3) and `workers` (at 51852c0e87, since v1.1) were added at too many years ago. So we need to pay for the compabitily.

But I can't imagine the 3rd party extensions touch these attributes...

Reply:
In my thoughts, nobody (or only very very less extensions) touches these attributes. So it's enough to add empty properties not to be crashed on updating. I know they will become not working if they really existed. But I can't imagine the extensions who install their custom workers or enqueue/dequeue to the checker queue.

Reply:
Can you explain the reason why empty properties are better than a crash?

IMHO, a crash is explicit about the incompatibility of an extension, where an empty property allows results to be silently incorrect.

Reply:
I realize there’ll be a `PendingDeprecationWarning`, which is ignored by default warning filters, so only users of the [Python development mode](https://docs.python.org/3/library/devmode.html#devmode) or who otherwise enabled these warnings see them.

Reply:
>Can you explain the reason why empty properties are better than a crash?
>IMHO, a crash is explicit about the incompatibility of an extension, where an empty property allows results to be silently incorrect.

What should users do when crashed? Extensions will not be updated as soon. I think "not working" is much better than crashes for them.

>I realize there’ll be a PendingDeprecationWarning, which is ignored by default warning filters, so only users of the Python development mode or who otherwise enabled these warnings see them.

Yes. `RemovedInSphinx50Warning` is a subclass of `PendingDeprecationWarning now. But it will be changed to a subclass of `DeprecationWarning` since 4.x series. Then it will be shown for users by default. As a result, they noticed their extensions (or hacks) uses deprecated APIs and internal functions.

Reply:
Finally, I determined to keep `builder.rqueue` and `builder.wqueue` as is (not removed, but deprecated) to merge this and step forward.